test/services/livesync/android-livesync-tool.ts

```diff
@@ -0,0 +1,533 @@
+			});
+
+			describe("sendFile", () => {
+				it("sends correct information", async () => {
+					//arrange
+					const filePath = path.join(testAppPlatformPath, rootJsFilePath);
+
+					//act
+					await livesyncTool.sendFile(filePath);
+
+					const sendFileData = getSendFileData((testSocket as TestSocket).accomulatedData);
+
+					//assert
+					assert.equal(sendFileData.fileContent, fileContents[rootJsFilePath]);
+					assert.equal(sendFileData.fileName, rootJsFilePath);
+					assert(sendFileData.headerHashMatch);
+					assert(sendFileData.fileHashMatch);
+					assert.equal(sendFileData.operation, AndroidLivesyncTool.CREATE_FILE_OPERATION);
+				});
+
+				it("rejects if file doesn't exist", () => {
+					//act
+					const sendFilePromise = livesyncTool.sendFile("nonexistent.js");
+
+					//assert
+					return assert.isRejected(sendFilePromise);
+				});
+
+				it("rejects if no connection", () => {
+					//arrange
+					livesyncTool.end();
+					const filePath = path.join(testAppPlatformPath, rootJsFilePath);
+
+					//act
+					const sendFilePromise = livesyncTool.sendFile(filePath);
+
+					//assert
+					return assert.isRejected(sendFilePromise, AndroidLivesyncTool.NO_SOCKET_CONNECTION_AVAILABLE_ERROR);
+				});
+
+				it("rejects if socket sends error", () => {
+					//arrange
+					const errorMessage = "Some error";
+					const filePath = path.join(testAppPlatformPath, rootJsFilePath);
+					testSocket.emit('error', errorMessage);
+
+					//act
+					const sendFilePromise = livesyncTool.sendFile(filePath);
+
+					//assert
+					return assert.isRejected(sendFilePromise, errorMessage);
+				});
+
+				it("rejects if error received", async () => {
+					//arrange
+					const filePath = path.join(testAppPlatformPath, rootJsFilePath);
+					const errorMessage = "Some error";
+					await livesyncTool.sendFile(filePath);
+					sandbox.stub(testSocket, "writeAsync").callsFake((data) => {
+						testSocket.emit('data', getSyncResponse(AndroidLivesyncTool.ERROR_REPORT, errorMessage));
+						return Promise.resolve();
+					});
+
+					//act
+					const sendFilePromise = livesyncTool.sendFile(filePath);
+
+					//assert
+					assert.isRejected(sendFilePromise, errorMessage);
+				});
+			});
+
+			describe("remove file", () => {
+				it("sends correct information", async () => {
+					//arrange
+					const filePath = path.join(testAppPlatformPath, rootJsFilePath);
+					await livesyncTool.removeFile(filePath);
+
+					//act
+					const removeData = getRemoveFileData((testSocket as TestSocket).accomulatedData);
+
+					//assert
+					assert.equal(removeData.fileName, rootJsFilePath);
+					assert.equal(removeData.operation, AndroidLivesyncTool.DELETE_FILE_OPERATION);
+					assert(removeData.headerHashMatch);
+				});
+			});
+
+			describe("sendDoSync", () => {
+				it("resolves after received data", () => {
+					//arrange
+					let doSyncResolved = false;
+
+					//act
+					const doSyncPromise = livesyncTool.sendDoSyncOperation(true);
```

Comment:
This is more readable 
```
livesyncTool.sendDoSyncOperation({ doRefresh: true })
```

Reply:
I think we have discussed this already. Having a definition of the method shows exactly how each parameter is called. If you still insist I can change it.